src/react/contexts/SpaceContext.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { useSpaces } from '../useSpaces.js';
```

Comment:
should this not be `useSpaces.js`?

Reply:
Sorry, I didn't get it, why not?

Reply:
the filename in the PR is `useSpace` not `useSpaces` it seems?

Reply:
It's provider from `useSpaces` we get current `spaces` instance. I agree it's a bit confusing, probably we can remove `SpacesProveder` and `useSpaces` and use `spaces` instance as property. What do you think?

Reply:
Sorry, just so I get this right, could you provide a short code example of the two different approaches?

Reply:
Current implementation:

```jsx
<SpacesProvider client={spacesInstance}>
    <Space name="space1">
           ...content
    </Space>
    <Space name="space2">
           ...content
    </Space>
</SpacesProvider>
```

I suggested:

```jsx
<Space name="space1" client={spacesInstance}>
      ...content
</Space>
<Space name="space2" client={spacesInstance}>
       ...content
</Space>
```